### DISCUSSION TITLE
supported ssh key types - 2048 vs 1024 bit ssh-dss keys

### DISCUSSION THREAD

**Discussion by AntonyKing7D** [drakkan/sftpgo#651] [Q&A] [answered]
Hi, I'm importing a number of users from proftpd into sftpgo ; most of them have the public key type ssh-rsa, but there are a significant number using ssh-dss . Although ssh-keygen -l sees the keys as valid, some ssh-dss keys are accepted and others are not . It looks like keys with 2048 bits are failing, but 1024 bits are ok. Is that expected?

**Comment by drakkan** [comment, marked as answer]
Hi,

it seems that crypto/ssh only support 1024 bit dss keys

https://github.com/golang/crypto/blob/master/ssh/keys.go#L411

I'll try to better investigate the next week. I have limited internet connection now

**Reply by AntonyKing7D** [reply]
Thanks for checking the library, that's really good to know. For our particular situation, I've just confirmed it's not a problem as it turns out that all those particular keys are no longer in use.

**Reply by AntonyKing7D** [reply]
I've since learned that 2048 bit DSA keys are not strictly valid anyway, so I'm happy to demand that our clients don't use them.
eg https://serverfault.com/questions/664676/ssh-with-dsa-2048-bits-key/664753

**Reply by drakkan** [reply]
> I've since learned that 2048 bit DSA keys are not strictly valid anyway, so I'm happy to demand that our clients don't use them. eg https://serverfault.com/questions/664676/ssh-with-dsa-2048-bits-key/664753

yes, this is the reason that in crypto/ssh there this comment:

> FIPS 186-3 allows for larger key sizes, which would confuse SSH

ssh-dss are obsolete, if you can avoid these keys is better. 

It is most interesting to support RSA-SHA2 (that should happen soon) for example
